### PR TITLE
refactor(automod ): Add new triggers, fix create rule endpoint

### PR DESCRIPTION
**Pull request by Damego**
## About

This pull request is about (X), which does (Y).

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [x] For the documentation
  - [x] To add a new feature
  - [x] As a general enhancement
  - [x] As a refactor of the library/the library's code
  - [x] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/api/models/misc.py [hunk 1]
```diff
@@ -242,11 +249,19 @@ class AutoModTriggerMetadata(DictSerializerMixin):
     A class object used to represent the trigger metadata from the AutoMod rule object.
 
     :ivar Optional[List[str]] keyword_filter: Words to match against content.
+    :ivar Optional[List[str]] regex_patterns: Regular expression patterns to match against content.
     :ivar Optional[List[str]] presets: The internally pre-defined wordsets which will be searched for in content.
+    :ivar Optional[List[int]] allow_list: Substrings which will be exempt from triggering the preset trigger type.
```
Comment (sillyElvy) [inline interactions/api/models/misc.py]: Strings but list of int?

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/models/misc.py (hunk 1 above)
Comment (Damego) [reply interactions/api/models/misc.py]: idk why im wrote like that

**Review by sillyElvy — CHANGES_REQUESTED** [review]
Merge conflict

**Comment by Damego** [comment]
Resolved

**Review by sillyElvy — APPROVED** [review]

**Pull request merged**